climate_index_collection/reductions.py

```diff
@@ -428,3 +503,97 @@ def eof_weights(dobj):
         Square root of weights needed to pre-process input data for SVD.
     """
     return np.sqrt(np.cos(np.deg2rad(dobj.coords["lat"])))
+
+
+def polygon_prime_meridian(pg):
+    """
+    Transforms shapely Polygons or MultiPolygons defined in [180W, 180E) coords into [0E,360E) coords.
+    Takes care of Polygons crossing the prime meridan.
+    Polygon points are expected be (lon, lat) tuples.
+
+    Parameters
+    ----------
+    pg: shaply Polygon or shapely MultiPolygon
+        Polygon including the area wanted.
+
+    Returns
+    -------
+    shapely MultPolygon
+        shaply MultiPolygon containing at least one Polygon.
+    """
+
+    # handle empty Polygons and MultiPolygons
+    if pg.is_empty:
+        return MultiPolygon([pg])
+
+    # create prime_meridian to eventually split the polygon
+    prime_meridian = LineString([(0, 90), (0, 0), (0, -90)])
+    pg_split = split(pg, prime_meridian)
+
+    # create a list containing all Polygons given by the split operation
+    # polygons on the negative (western) side of the prime meridian are translated into new coords, by adding 360 to the lon values.
+    pg_list = []
+    for temp_pg in pg_split.geoms:
+        # check if the polygons minx is negative and add 360 to it.
+        if temp_pg.bounds[0] < 0:
+            temp_pg = translate(temp_pg, xoff=360)
+        pg_list += [temp_pg]
+
+    # create the multipolygon existing in [0E, 360E) coords from the list of polygons
+    result = unary_union(pg_list)
+
+    # for consistency always return MultiPolygon
+    if type(result) is not MultiPolygon:
+        # convert Polygon to MultiPolygon
+        return MultiPolygon([result])
+    else:
+        return result
+
+
+def polygon2mask(dobj, pg, lat_name="lat", lon_name="lon"):
+    """
+    This funciton creates a mask for a given DataArray or DataSet based on a shapely Polygon or MultiPolygon.
+    Polygon points are expected be (lon, lat) tuples.
+    The dobj is expected to have lon values in [0E, 360E) coords.
+
+    Parameters
+    ----------
+    dobj: xarray.Dataset or xarray.DataArray
+        Contains the original data.
+    pg: shapely Polygon or shapely MultiPolygon
+        Polygon including the area wanted.
+    lat_name: str
+        Name of the latitude coordinate. Defaults to "lat".
+    lon_name: str
+        Name of the longitude coordinate. Defaults to "lon".
+
+    Returns
+    -------
+    xarray.Dataset or xarray.DataArray
+        Mask for given Polygon on dobj grid.
+    """
+
```

Comment:
Probably something to handle the behaviour I mentioned above:

```suggestion
If pg is None:
   return xr.DataArray(
       np.full(np.shape(lon_2d), True),
       dims=lon_2d.dims,
       coords=lon_2d.coords
      )
```

Might be a solution for this other comment.
I had the idea to only return True, but this would make things uncomfortable if one expects a xr.DataArray.
There might be a better way to create a fully True array of shape lon_2d.